packages/headless/src/modules/manufacturer/graphql/manufacturer.resolver.ts

```diff
@@ -1,71 +1,93 @@
-import { UpdateManufacturerDto } from './../dto/updateManufacturer.dto';
-import { GetManufacturersQueryDto } from './../dto/getManufacturers.dto';
-import { CreateManufacturerDto } from '../dto/createManufacturer.dto';
-import { Manufacturer } from 'src/entity/manufacturer';
 import { ManufacturerService } from './../services/manufacturer.service';
 import { Args, Mutation, Query, Resolver } from '@nestjs/graphql';
 import { UseGuards } from '@nestjs/common';
 import { RolesGuard } from 'src/guards/auth.guard';
+import {
+  AllManufacturersResponse,
+  ManufacturerInput,
+  ManufacturerResponse,
+  ManufacturerSchemaGql,
+  ManufacturersQuery,
+} from './manufacturer.model';
+import { Helper } from 'src/helper/helper.interface';
 
 @UseGuards(new RolesGuard(['admin']))
-@Resolver()
+@Resolver((of) => ManufacturerSchemaGql)
 export class ManufacturerResolver {
-  constructor(private manufacturerService: ManufacturerService) { }
+  constructor(
+    private manufacturerService: ManufacturerService,
+    private helper: Helper,
+  ) {}
 
-  /**
-   * @Graphql
-   * Query getAllManufacturers calls form manufacturer.graphql file
-   * @param skip Optional
-   * @param limit Optional
-   * @returns {Object} Object { error code data}
-   */
-  @Query()
-  async getAllManufacturers(@Args() query: GetManufacturersQueryDto) {
-    return await this.manufacturerService.getAllManufacturers(query);
+  @Query(() => AllManufacturersResponse, {
+    name: 'getAllManufacturers',
+    description:
+      'Get all manufacturers passing with optional query as skip and limit',
+  })
+  async getAllManufacturers(
+    @Args({
+      name: 'query',
+      type: () => ManufacturersQuery,
+      nullable: true,
+    })
+    query?: ManufacturersQuery,
+  ) {
+    const res = await this.manufacturerService.getAllManufacturers(query);
+    return this.helper.serviceResponse.graphqlResponse(res);
   }
 
-  /**
-   * @Graphql
-   * Query getManufacturer calls form manufacturer.graphql file
-   * @param manufacturerId 
-   * @returns {Object} Object { error code data}
-   */
-  @Query()
-  async getManufacturer(@Args('manufacturerId') manufacturerId: string) {
-    return await this.manufacturerService.getManufacturer(manufacturerId);
+  @Query(() => ManufacturerResponse, {
+    name: 'getManufacturer',
+    description: 'Get single manufacturer by id',
+  })
+  async getManufacturer(
+    @Args({ name: 'manufacturerId', type: () => String })
```

Comment:
don't pass the name additionally. because as far i know this name will be used as the input property name. but if you didn't pass the name then the parameter name would have been the graphql input name. if you have different thoughts then share

Reply:
Yeah, got it. It should be used under Mutations where input is passed.